src/components/Editor/Breakpoint.js

```diff
@@ -34,39 +35,29 @@ class Breakpoint extends Component {
     this.addBreakpoint = this.addBreakpoint.bind(this);
   }
 
-  isGeneratedSource() {
-    return isGeneratedId(this.props.selectedSource.get("id"));
-  }
-
-  getLocation() {
-    const { breakpoint } = this.props;
-    return this.isGeneratedSource()
-      ? breakpoint.generatedLocation
-      : breakpoint.location;
-  }
-
   addBreakpoint() {
-    const { breakpoint, editor } = this.props;
+    const { breakpoint, editor, selectedSource } = this.props;
 
     // NOTE: we need to wait for the breakpoint to be loaded
     // to get the generated location
-    if (this.isGeneratedSource() && breakpoint.loading) {
+    if (!selectedSource || breakpoint.loading) {
       return;
     }
 
-    const location = this.getLocation();
-    const line = location.line - 1;
+    const line = breakpoint.location.line - 1;
 
-    editor.setGutterMarker(
+    showSourceText(editor, selectedSource.toJS());
```

Comment:
This is important for making sure that we're updating the correct codemirror document.